Move to gradle

anghelutar opened the pull request:
Attention points:

1) Current Alfresco images have already a jodconverter amp included in the war. 
In order for that to be removed, set leanImage = false when building the image.

2) Current Alfresco images use defaults good for the old remote jodconverter. New endpoint needs to be overwritten via an environment variable in docker-compose:
GLOBAL_content.transformer.RemoteJODConverter.endpoint=http://jodconverter:8080/converter

Commit:
JODCONV-4: move build to gradle

vierbergenlars commented on build.gradle:
```diff
@@ -0,0 +1,103 @@
+    leanImage = false
+
+    dockerBuild {
+        repository = 'xeniteu/alfresco-remote-jodconverter'
```
We don't want to push any alfresco image, so no need to give a full repository here (you can just call it `alfresco-remote-jodconverter`)

vierbergenlars commented on build.gradle:
```diff
@@ -0,0 +1,103 @@
+        }
+        maven {
+            name 'release'
+            url 'https://artifactory.xenit.eu/artifactory/libs-release-local'
```
Remove this release repository

vierbergenlars commented on build.gradle:
```diff
@@ -0,0 +1,103 @@
+
+
+dockerAlfresco {
+    baseImage = "hub.xenit.eu/alfresco-community:${alfrescoVersion}"
```
This image is not available publicly

vierbergenlars replied:
You can base it on a skeleton image, then there is no alfresco war in there yet

anghelutar replied:
I added a comment that it will need to change with a public image, once it is available.

Skeleton images do not have any connector incorporated, that's why I used a community image + leanImage=false

vierbergenlars commented on src/test/resources/docker-compose.yml:
```diff
@@ -0,0 +1,26 @@
+#      - CATALINA_OPTS=-Xdebug -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8000
+
+  postgresql:
+    image: hub.xenit.eu/postgres
```
Use `xeniteu/postgres`

Commit:
Use a public image for postgres.
Alfresco images are not yet public - added comment about that.

Commit:
Fix warnings because of missing info for javadocs.

Commit:
Publishing to maven central:
1) release branches published to releases, other branches to snapshots
2) signing required for releases
3) publishing and signing variables passed via environment variables

Commit:
Travis file to publish automatically.
For the moment - only snapshots.
Releases to be done manually.

Commit:
Added some words about publishing to the README.

Commit:
Added Alfresco public repo

Commit:
Bumped version to 2.0.0.
Signing parameters outside the publishing section.

Commit:
Only sign if sign variables are set.

Commit:
Prepare for a release to maven central: add encrypted file for gpg key.

Commit:
Re-encrypt variables or add them in clear to the travis file.

Commit:
Before install commands in one place. Key needs only the last 8 characters.

Commit:
Sign snapshots as well.

Commit:
Added vsdx format.
